mob_test.go

```diff
@@ -742,6 +759,95 @@ func TestStartOnUnpushedFeatureBranch(t *testing.T) {
+	assertOutputContains(t, output, "mob start --create")
+}
+
+func TestStartOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	output, configuration := setup(t)
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOnBranch(t, "feature1")
+	assertOutputContains(t, output, "mob start --include-uncommitted-changes")
+}
+
+func TestStartCreateOnUnpushedFeatureBranch(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	output, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOutputContains(t, output, "To start, including uncommitted changes, use")
+	assertOutputContains(t, output, "mob start --create --include-uncommitted-changes")
+}
+
+func TestStartCreateIncludeUncommitedChangesOnUnpushedFeatureBranchWithUncommitedChanges(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	configuration.StartIncludeUncommittedChanges = true
+	git("checkout", "-b", "feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnPushedFeatureBranch(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	checkoutBranchAndCreateRemoteBranch("feature1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+}
+
+func TestStartCreateOnPushedFeatureBranchWhichIsAhead(t *testing.T) {
+	_, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	checkoutBranchAndCreateRemoteBranch("feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	git("add", ".")
+	git("commit", "-m", "commit ahead")
+	git("push")
+	git("reset", "--hard", "HEAD~1")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "mob/feature1")
+	assertCommitLogContainsMessage(t, "mob/feature1", "commit ahead")
+}
+
+func TestStartCreateOnPushedFeatureBranchWhichIsBehind(t *testing.T) {
+	output, configuration := setup(t)
+	configuration.StartCreateRemoteBranch = true
+	checkoutBranchAndCreateRemoteBranch("feature1")
+	assertOnBranch(t, "feature1")
+	createFile(t, "file.txt", "contentIrrelevant")
+	git("add", ".")
+	git("commit", "-m", "commit ahead")
+	assertOnBranch(t, "feature1")
+	start(configuration)
+
+	assertOnBranch(t, "feature1")
+	assertOutputContains(t, output, "ERROR cannot start; unpushed changes on base branch must be pushed upstream")
```

Comment:
'unpushed changes on base branch' is sufficient in my opinion. How the user wants to solve this problem is their decision.

Reply:
I could change it, but thats the behavior it hab before ;)